tools/c7n_azure/c7n_azure/handler.py

```diff
@@ -52,7 +53,10 @@ def run(event, context):
     policies = PolicyCollection.from_data(policy_config, options)
     if policies:
         for p in policies:
-            p.push(event, context)
+            try:
+                p.push(event, context)
+            except CloudError as error:
+                log.warning("Failed to process policy: %s :: %s" % (p.name, error))
```

Comment:
log.error? 
And what about HttpError? (similar to what Stefan did in his PR)

Reply:
AzureHttpError comes from AzureException comes from Exception  -- it seems one set of errors comes from AutoRest and another set from other parts of the SDK.  I don't fully grok.

https://github.com/Azure/azure-sdk-for-python/blob/master/azure-common/azure/common/__init__.py

Reply:
I also didn't really understand why we want to not fail?  If we let the function fail then don't we get better metrics/analytics and stuff automatically on the function health?  Are we doing something now where we do multiple runs in a single function event?

Reply:
i'm curious about the same. if we're monitoring the function, this seems like it obscures the ability to detect an issue.